controllers/managedfusionoffering_controller.go

```diff
@@ -147,15 +180,14 @@ func (r *ManagedFusionOfferingReconciler) reconcilePhases() (reconcile.Result, e
 		if err := r.reconcileSubscription(); err != nil {
 			return ctrl.Result{}, err
 		}
+	}
 
-		if result, err := pluginReconcile(r, r.managedFusionOffering); err != nil {
-			return ctrl.Result{}, fmt.Errorf("an error was encountered during reconcile: %v", err)
-		} else {
-			return result, nil
-		}
+	if result, err := pluginReconcile(r, r.managedFusionOffering); err != nil {
+		return ctrl.Result{}, fmt.Errorf("an error was encountered during reconcile: %v", err)
+	} else {
+		return result, nil
 	}
```

Comment:
is there no complaint from any linter that the function is missing final return?

Reply:
No complaints from any linter. We have return in the else so linter does not see it as an issue.